gqcp/include/Basis/Transformations/SimpleOrbitalRotationGenerators.hpp

```diff
@@ -0,0 +1,149 @@
+ *   
+ *  This class is used as a base class for `ROrbitalRotationGenerator` and `GOrbitalRotationGenerator`, since they are both expressed using a single vector, as opposed to `UOrbitalRotationGenerator`, which uses separate kappa vectors for alpha- and beta- generators. The word 'simple' is used here as an antonym for 'compound'.
+ * 
+ *  @tparam _Scalar                                           The scalar type used for a orbital rotation generator vector: real or complex.
+ *  @tparam _DerivedOrbitalRotationGenerator                  The type of the orbital rotation generator that derives from this class, enabling CRTP and compile-time polymorphism.
+ */
+template <typename _Scalar, typename _DerivedOrbitalRotationGenerators>
+class SimpleOrbitalRotationGenerators {
+
+public:
+    // The scalar type used for a orbital rotation generator vector: real or complex.
+    using Scalar = _Scalar;
+
+    // The type of the orbital rotation generator that derives from this class, enabling CRTP and compile-time polymorphism.
+    using DerivedOrbitalRotationGenerators = _DerivedOrbitalRotationGenerators;
+
+private:
+    // The number of spatial orbitals that can be rotated using these orbital rotation generators.
+    size_t number_of_spatial_orbitals;
+
+    // The strict upper/lower triangle of the kappa matrix.
+    VectorX<Scalar> kappa_vector;
+
+
+public:
+    /**
+     *  MARK: Constructors
+     */
+
+    /**
+     *  Create a `SimpleOrbitalRotationGenerator' from a given kappa vector.
+     * 
+     *  @param  kappa_vector        The orbital rotation generators represented as a vector that corresponds to the strict upper/lower triangle of the kappa matrix.
+     */
+    SimpleOrbitalRotationGenerators(const VectorX<Scalar>& kappa_vector) :
+        kappa_vector {kappa_vector},
+        number_of_spatial_orbitals {strictTriangularRootOf(kappa_vector.size())} {}
+
+
+    /**
+     *  Create a `SimpleOrbitalRotationGenerator' from a given kappa matrix.
+     * 
+     *  @param  kappa_matrix        The orbital rotation generators represented as a vector that corresponds to the full antisymmetric the kappa matrix.
+     */
+    SimpleOrbitalRotationGenerators(const SquareMatrix<Scalar>& kappa_matrix) :
+        SimpleOrbitalRotationGenerators(kappa_matrix.pairWiseStrictReduced()) {}
+
+
+    /**
+     *  MARK: Named Constructors
+     */
+
+    /**
+     *  Construct orbital rotation generators by adding redundant (i.e. 0) occupied-virtual and virtual-virtual generators to the given occupied-occupied generators.
+     * 
+     *  @param o_o_generators       The occupied-occupied orbital rotation generators.
+     *  @param K                    The total number of spatial orbitals.
+     * 
+     *  @return The 'full' orbital rotation generators from the given occupied-occupied generators.
+     */
+    static DerivedOrbitalRotationGenerators FromOccOcc(const DerivedOrbitalRotationGenerators& o_o_generators, const size_t K) {
```

Comment:
What about:
- improving this API to `FromOccupationTypes` that accepts `(generators, row_occupation_type, column_occupation_type, orbital_space)`, using `GQCP::OccupationType`?
- adding this API to `UOrbitalRotationGenerators` as well?

Reply:
Is this the API you had in mind?

```c++
/**
     *  Construct orbital rotation generators by adding redundant (i.e. 0) generators to the given occupation_type - occupation_type generators.
     * 
     *  @param generators                           The orbital rotation generators of the specified ocupation types.
     *  @param row_occupation_type                  The occupation type of the rows of the orbital rotation generator kappa matrix.
     *  @param column_occupation_type               The occupation type of the column of the orbital rotation generator kappa matrix.
     *  @param orbital_space                        The total orbital space in which the orbitals are rotated.
     * 
     *  @return The 'full' orbital rotation generators from the given row_occupation_type - column_occupation_type generators.
     */
    static DerivedOrbitalRotationGenerators FromOccupationTypes(const DerivedOrbitalRotationGenerators& generators, const OccupationType row_occupation_type, const OccupationType column_occupation_type, const OrbitalSpace& orbital_space) {

        // Use the orbital space to determine the size of the full kappa matrix.
        SquareMatrix<Scalar> kappa_full_matrix = SquareMatrix<Scalar>::Zero(orbital_space.numberOfOrbitals());

        // Depending on the row and column occupation types, we fill in the correct block of the total kappa matrix and leave the rest to be zero.
        if (row_occupation_type == OccupationType::k_occupied && column_occupation_type == OccupationType::k_occupied) {
            kappa_full_matrix.topLeftCorner(generators.numberOfOrbitals(), generators.numberOfOrbitals()) = generators.asMatrix();
        } else if (row_occupation_type == OccupationType::k_occupied && column_occupation_type == OccupationType::k_virtual) {
            kappa_full_matrix.topRightCorner(generators.numberOfOrbitals(), generators.numberOfOrbitals()) = generators.asMatrix();
        } else if (row_occupation_type == OccupationType::k_virtual && column_occupation_type == OccupationType::k_occupied) {
            kappa_full_matrix.BottomLeftCorner(generators.numberOfOrbitals(), generators.numberOfOrbitals()) = generators.asMatrix();
        } else {
            kappa_full_matrix.BottomRightCorner(generators.numberOfOrbitals(), generators.numberOfOrbitals()) = generators.asMatrix();
        }

        return DerivedOrbitalRotationGenerators(kappa_full_matrix);
    }
```

Reply:
Yep, that's the one. I think I went a little too far by think we would have to supply an `OrbitalSpace`, so you can just accept a `size_t` that represents the total number of orbitals.